lightning/src/ln/data_persister.rs

```diff
@@ -0,0 +1,70 @@
+//! Logic for persisting data from ChannelMonitors on-disk. Sample data
+//! persisters are separated into the lightning-persist-data crate. These
+//! objects mainly interface with the ChainMonitor when a channel monitor is
+//! added or updated, and when they are all synced on startup.
+use chain::channelmonitor::{ChannelMonitor, ChannelMonitorUpdate, ChannelMonitorUpdateErr};
+use chain::keysinterface::ChannelKeys;
+use chain::transaction::OutPoint;
+
+/// ChannelDataPersister is responsible for persisting channel data: this could
+/// mean writing once to disk, and/or uploading to one or more backup services.
+///
+/// Note that for every new monitor, you **must** persist the new ChannelMonitor
+/// to disk/backups. And, on every update, you **must** persist either the
+/// ChannelMonitorUpdate or the updated monitor itself. Otherwise, there is risk
+/// of situations such as revoking a transaction, then crashing before this
+/// revocation can be persisted, then unintentionally broadcasting a revoked
+/// transaction and losing money. This is a risk because previous channel states
+/// are toxic, so it's important that whatever channel state is persisted is
+/// kept up-to-date.
+///
+/// Given multiple backups, situations may arise where one or more backup sources
+/// have fallen behind or disagree on the current state. Because these backup
+/// sources don't have any transaction signing/broadcasting duties and are only
+/// supposed to be persisting bytes of data, backup sources may be considered
+/// "in consensus" if a majority of them agree on the current state and are on
```

Comment:
@ariard @devrandom FYI because it was your guys initial idea to specify to users that the majority is OK for backup consensus -- I'm gonna replace the "majority consensus" docs with these nuances^.

Reply:
Ok the docs I ended up writing might be too obvious. I'll look again tmrw but here they are: https://github.com/rust-bitcoin/rust-lightning/pull/681/commits/d6c13126f4c2b1e7d6f0585a9ded4e06a5d81cfb#diff-327c087952e34acbde50a6f89a6a97d8R2121

Reply:
The linked doc looks good to me.

Reply:
See https://github.com/rust-bitcoin/rust-lightning/pull/681#discussion_r498453411